rust/build.rs

```diff
@@ -0,0 +1,16 @@
+use cfg_aliases::cfg_aliases;
+
+fn main() {
+    // Setup cfg aliases
+    cfg_aliases! {
+        // OS-exclusive syscalls
+        chflags: { file_flags },
+        lchmod: { any(target_os = "netbsd", target_os = "freebsd", target_os = "dragonfly") },
+        lchflags: { any(target_os = "openbsd", target_os = "netbsd", target_os = "freebsd",
+                    target_os = "dragonfly", target_os = "macos", target_os = "ios") },
+        // OS-exclusive features
+        file_flags: { any(target_os = "openbsd", target_os = "netbsd", target_os = "freebsd",
+                    target_os = "dragonfly", target_os = "macos", target_os = "ios") },
+        birthtime: { any(target_os = "freebsd", target_os = "ios", target_os = "macos", target_os = "netbsd", target_os = "openbsd") }
+    }
+}
```

Comment:
You need to tell Cargo to expect these cfg aliases, or else it will warn about unknown cfg expressions.
```suggestion
    println!("cargo:rustc-check-cfg=cfg(chflags)");
    println!("cargo:rustc-check-cfg=cfg(lchmod)");
    println!("cargo:rustc-check-cfg=cfg(lchflags)");
    println!("cargo:rustc-check-cfg=cfg(file_flags)");
    println!("cargo:rustc-check-cfg=cfg(birthtime)");
}
```

Reply:
It's already done by cfg_aliases.

Reply:
Ahh, sweet.  That looks like a new feature of cfg_aliases 0.2.1, which is newer than the last version I used.  Thanks for the tip.  I've got another project to update now.